### PR TITLE
Add pop_twice_to_wake and pop_twice_to_repeat functions

### PR DESCRIPTION
**Pull request by trillium**
Following on from #1364

Haven't seen that PR change much over the past few weeks, I've been using that functionality in my personal repo and it's been great so I wanted to make sure other users got to try it out. 

## Changes from #1364

 - Switched to tag at request of @pokey in review
   - This will prevent blocking on other pop commands
 - Added Module import to be able to define tag
 - Updated context matcher to look for `user.pop_twice_to_wake`
 - Dropped changes to mouse.py 

---

 - Merged with #1399 as per discussion in grooming session
 - Add explanation comments
 - Switch double pop time minimum  and maximum to a user.setting

### CODE REVIEW

**Comment by pokey** [comment]
I think I would lean towards just merging this PR into https://github.com/talonhub/community/pull/1399. It will make the code duplication more obvious. I don't feel strongly tho, and I do generally like small, atomic PRs

**Comment by Spiteless** [comment]
Happy to merge into #1399 if that is easier

**Comment by nriley** [comment]
- Please add some comments indicating that if you enable either of these tags, that it will disable the existing pop to click or scroll functionality from the existing community code.
- Please add an inter-pop adjustment setting just in case people can pop more frequently than every 0.1 seconds

**Comment by Spiteless** [comment]
Addtional request from grooming session was to merge the two PR's into one ( #1399 )

File: core/modes/sleep_mode_pop_twice_to_wake.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+from talon import Context, Module, actions, settings
+
+ctx = Context()
+mod = Module()
+
+mod.tag("pop_twice_to_wake", desc="tag for enabling pop twice to wake in sleep mode")
+
+mod.setting(
+    "last_pop_time_maximum",
```
Comment (pokey) [inline core/modes/sleep_mode_pop_twice_to_wake.py:11]: I think I'd lean towards something like

```suggestion
    "double_pop_speed_minimum",
```

File: core/modes/sleep_mode_pop_twice_to_wake.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    type=float,
+    desc="""Shortest time in seconds to accept a second pop to trigger additional actions""",
+    default=0.1,
+)
+
+mod.setting(
+    "last_pop_time_maximum",
```
Comment (pokey) [inline core/modes/sleep_mode_pop_twice_to_wake.py:18]: and then

```suggestion
    "double_pop_speed_maximum",
```

File: core/modes/sleep_mode_pop_twice_to_wake.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+ctx.matches = r"""
+mode: sleep
+and tag: user.pop_twice_to_wake
+"""
+
+time_last_pop = 0
+last_pop_time_minimum = settings.get(
+    "user.mouse_enable_pop_stops_scroll"
```
Comment (pokey) [inline core/modes/sleep_mode_pop_twice_to_wake.py:31]:
```suggestion
double_pop_speed_minimum = settings.get(
    "user.double_pop_speed_minimum"
```

File: core/modes/sleep_mode_pop_twice_to_wake.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+)  # default 0.1
+last_pop_time_maximum = settings.get(
+    "user.mouse_enable_pop_stops_scroll"
```
Comment (pokey) [inline core/modes/sleep_mode_pop_twice_to_wake.py:34]:
```suggestion
double_pop_speed_maximum = settings.get(
    "user.double_pop_speed_maximum"
```

File: plugin/repeater/pop_twice_to_repeat.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+and tag: user.pop_twice_to_repeat
+"""
+
+time_last_pop = 0
+last_pop_time_minimum = settings.get(
+    "user.mouse_enable_pop_stops_scroll"
+)  # default 0.1
+last_pop_time_maximum = settings.get(
+    "user.mouse_enable_pop_stops_scroll"
+)  # defualt 0.3
```
Comment (pokey) [inline plugin/repeater/pop_twice_to_repeat.py:21]:
```suggestion
double_pop_speed_minimum = settings.get(
    "user.double_pop_speed_minimum"
)  # default 0.1
double_pop_speed_maximum = settings.get(
    "user.double_pop_speed_maximum"
)  # defualt 0.3
```

File: settings.talon [hunk 6]
```diff
@@ -47,6 +47,10 @@ settings():
     # Set the total number of command history lines to display
     user.command_history_size = 50
 
+    # Set the time window size for to for pop_twice_to_sleep and pop_twice_to_repeat
+    user.last_pop_time_manimum = 0.1
+    user.last_pop_time_maximum = 0.3
```
Comment (pokey) [inline settings.talon:52]:
```suggestion
    # Set the time window size for to for pop_twice_to_sleep and pop_twice_to_repeat. By default, the pops must be more than 0.1 seconds apart and less then 0.3 seconds, to reduce false positives
    user.double_pop_speed_minimum = 0.1
    user.double_pop_speed_maximum = 0.3
```

File: settings.talon [hunk 7]
```diff
@@ -76,3 +80,14 @@ settings():
 # prone to false positives. If you uncomment this, you will need to say
 # "numb one" to write "1". Note that this tag will eventually be activated by default
 # tag(): user.prefixed_numbers
+
+# Uncomment below enable pop_twice_to_wake
+# Without this tag noise_trigger_pop is usually associated with pop to click actions
+# Enabling this tag disables other pop to click actions in sleep mmode, including pop to click
```
Comment (pokey) [inline settings.talon:86]:
```suggestion
# Enabling this tag disables other pop to click actions in sleep mode, including pop to click
```

File: settings.talon [hunk 8]
```diff
@@ -76,3 +80,14 @@ settings():
...
+# tag(): user.pop_twice_to_wake
+
+# Uncomment below enable pop_twice_to_repeat
+# Enabling this tag will repeat the last command when two pops are heard within the alloted time window
```
Comment (pokey) [inline settings.talon:90]:
```suggestion
# Enabling this tag will repeat the last command when two pops are heard within the allotted time window
```

**Review by pokey — COMMENTED** [review]
I made some naming tweaks / fixed some things. We should give this thing a test locally before merging too

**Comment by Spiteless** [comment]
Accepted changes from @pokey

**Comment by Spiteless** [comment]
Just tested locally, works for both sleep and repeat

**Review by AndreasArvidsson — APPROVED** [review]

**Comment by knausj85** [comment]
I believe this is ready to be merged. As long as there are no objections, I will do so this weekend

**Review by knausj85 — APPROVED** [review]

**Pull request merged**